contracts/modules/TransferManager/GTM/GeneralTransferManager.sol

```diff
@@ -260,57 +264,57 @@ contract GeneralTransferManager is GeneralTransferManagerStorage, TransferManage
-    * @param _toTime is the moment when the purchase lockup period ends and the investor can freely purchase tokens from others
+    * @param _canSendAfter is the moment when the sale lockup period ends and the investor can freely sell or transfer their tokens
+    * @param _canReceiveAfter is the moment when the purchase lockup period ends and the investor can freely purchase or receive tokens from others
     * @param _expiryTime is the moment till investors KYC will be validated. After that investor need to do re-KYC
     */
     function modifyKYCData(
         address _investor,
-        uint256 _fromTime,
-        uint256 _toTime,
-        uint256 _expiryTime
+        uint64 _canSendAfter,
+        uint64 _canReceiveAfter,
+        uint64 _expiryTime
     )
         public
         withPerm(ADMIN)
     {
-        _modifyKYCData(_investor, _fromTime, _toTime, _expiryTime);
+        _modifyKYCData(_investor, _canSendAfter, _canReceiveAfter, _expiryTime);
     }
 
-    function _modifyKYCData(address _investor, uint256 _fromTime, uint256 _toTime, uint256 _expiryTime) internal {
+    function _modifyKYCData(address _investor, uint64 _canSendAfter, uint64 _canReceiveAfter, uint64 _expiryTime) internal {
         require(_investor != address(0), "Invalid investor");
         IDataStore dataStore = getDataStore();
         if (!_isExistingInvestor(_investor, dataStore)) {
            dataStore.insertAddress(INVESTORSKEY, _investor);
         }
-        uint256 _data = VersionUtils.packKYC(uint64(_fromTime), uint64(_toTime), uint64(_expiryTime), uint8(1));
+        uint256 _data = VersionUtils.packKYC(_canSendAfter, _canReceiveAfter, _expiryTime, uint8(1));
         dataStore.setUint256(_getKey(WHITELIST, _investor), _data);
-        emit ModifyKYCData(_investor, msg.sender, _fromTime, _toTime, _expiryTime);
+        emit ModifyKYCData(_investor, msg.sender, _canSendAfter, _canReceiveAfter, _expiryTime);
     }
 
     /**
     * @notice Add or remove KYC info of an investor.
     * @param _investors is the address to whitelist
-    * @param _fromTime is the moment when the sale lockup period ends and the investor can freely sell his tokens
-    * @param _toTime is the moment when the purchase lockup period ends and the investor can freely purchase tokens from others
+    * @param _canSendAfter is the moment when the sale lockup period ends and the investor can freely sell his tokens
+    * @param _canReceiveAfter is the moment when the purchase lockup period ends and the investor can freely purchase tokens from others
     * @param _expiryTime is the moment till investors KYC will be validated. After that investor need to do re-KYC
     */
     function modifyKYCDataMulti(
         address[] memory _investors,
-        uint256[] memory _fromTime,
-        uint256[] memory _toTime,
-        uint256[] memory _expiryTime
+        uint64[] memory _canSendAfter,
```

Comment:
in 2.1.0 you used `uint256` in function arguments and then typecast those into uint64 in the function body. Shouldn't we make consistency in both branches

Reply:
We used `uint256` in 2.1 to make the interface compatible with 2.0.

There is no need to keep 3.0 and 2.1 compatible.